components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc

```diff
@@ -19,23 +19,23 @@ The following formula describes the overall fee for a transaction:
 
 [stem]
 ++++
-F = \text{gas_price}\cdot\left(\max_k v_k w_k + 0.9\cdot\text{calldata_cost}\cdot\left((n+2m-1) + 3t + \sum\limits_{i=1}^t q_i +3\ell\right)+ t\cdot\text{storage_write_cost} + n \cdot 200 + 272\right)
+F = \text{gas_price}\cdot\left(\max_k v_k w_k + \text{calldata_cost}\cdot\left(2n+2(m-2) + 3t + \sum\limits_{i=1}^t q_i +3\ell\right) - \text{contract_update_discount}\cdot n + 310 + \text{storage_write_cost}\cdot t\right)
 ++++
 
 where:
 
-* stem:[$v$] is a vector that represents resource usage, where each of its entries, stem:[$v_k$], correspond to different resource types: Cairo steps and number of applications of each builtin.
+* stem:[$v$] is a vector that represents resource usage, where each of its entries, stem:[$v_k$], corresponds to different resource types: Cairo steps and number of applications of each builtin.
 +
 For more information see xref:#calculation_of_computation_costs[Calculation of computation costs].
 * stem:[$n$] is xref:#storage_updates[the number of unique contracts updated].
-* stem:[$m$] is the number of values updated, not counting multiple updates for the same key.
+* stem:[$m$] is the number of values updated, not counting multiple updates for the same key. Note that stem:[$m\ge 2$] always holds since the sending account balance and the sequencer's balance are always updated.
 * stem:[$t$] is the number of L2->L1 messages sent, where the corresponding payload sizes are denoted by stem:[$q_1,...,q_t$].
 * stem:[$\ell$] is the number of contract deployments.
-* stem:[$0.9$] is a 10% cost reduction that considers that the sequencer does not incur additional costs for repeated updates to the same storage slot within a single block.
 * stem:[$w$] is the xref:#calculation_of_computation_costs[`CairoResourceFeeWeights`] vector.
-* stem:[$\text{calldata_cost}$] is 612 gas per 32-byte word (~100 gas is charged for on-chain hashing the happens for every sent word).
+* stem:[$\text{calldata_cost}$] is 550 gas per 32-byte word. This cost is derived as follows: 512 gas per 32-byte word for calldata, ~100gas for on-chain hashing that happens for every sent word, and 10% discount because the sequencer does not incur additional costs for repeated updates to the same storage slot within a single block, resulting in 0.9*612 = 550gas (rounded down).
```

Comment:
```suggestion
* stem:[$\text{calldata_cost}$] is 550 gas per 32-byte word. This cost is derived as follows: 512 gas per 32-byte word for calldata, ~100 gas for onchain hashing that happens for every word sent, and a 10% discount that applies because the sequencer does not incur additional costs for repeated updates to the same storage slot within a single block. So the gas cost is calculated as follows: stem:[0.9 \cot 612 = 550 gas], rounded down.
```